src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.cs

```diff
@@ -48,14 +48,14 @@ public static partial class JsonSerializer
                         break;
                 }
 
-                if (flushThreshold >= 0 && writer.BytesPending > flushThreshold)
+                if (finishedSerializing && writer.CurrentDepth == 0)
                 {
-                    return false;
+                    continueWriting = false;
                 }
 
-                if (finishedSerializing && writer.CurrentDepth == 0)
+                if (continueWriting && flushThreshold >= 0 && writer.BytesPending > flushThreshold)
```

Comment:
What is this condition doing? Can you please add a comment since the condition is non-trivial.

Reply:
done